suites/message/nested.go

```diff
@@ -33,61 +35,235 @@ func TestNestedSends(t *testing.T, factory state.Factories) {
 		WithDefaultGasPrice(big.NewInt(1)).
 		WithActorState(drivers.DefaultBuiltinActorsState)
 
-	t.Run("ok basic send", func(t *testing.T) {
+	t.Run("ok basic", func(t *testing.T) {
 		td := builder.Build(t)
 		defer td.Complete()
 
-		_, aliceId := td.NewAccountActor(drivers.SECP, acctDefaultBalance)
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
 
-		stage := prepareStage(td, aliceId, multisigBalance, 0)
-		balanceBefore := td.GetBalance(aliceId)
+		// Multisig sends back to the creator.
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(stage.creator, amtSent, builtin.MethodSend, nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+
+		td.AssertActor(stage.creator, big.Sub(big.Add(balanceBefore, amtSent), recpt.GasUsed), 2)
+	})
+
+	t.Run("ok to new actor", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
+
+		// Multisig sends to new address.
+		newAddr := td.Wallet().NewSECP256k1AccountAddress()
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(newAddr, amtSent, builtin.MethodSend, nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+
+		td.AssertBalance(stage.msAddr, big.Sub(multisigBalance, amtSent))
+		td.AssertBalance(stage.creator, big.Sub(balanceBefore, recpt.GasUsed))
+		td.AssertBalance(newAddr, amtSent)
+	})
+
+	t.Run("ok to new actor with invoke", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
+
+		// Multisig sends to new address and invokes pubkey method at the same time.
+		newAddr := td.Wallet().NewSECP256k1AccountAddress()
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(newAddr, amtSent, builtin.MethodsAccount.PubkeyAddress, nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+		expected := bytes.Buffer{}
+		require.NoError(t, newAddr.MarshalCBOR(&expected))
+		assert.Equal(t, expected.Bytes(), recpt.ReturnValue)
```

Comment:
How does this work? My understanding is that the result of calling MethodsAccount.PubkeyAddress will be given to the multisig actor's Propose() method. However, Propose() will return nil, and so this address won't actually get put into the message receipt. Is that wrong?

Reply:
@arajasek this reflects my understanding as well, the Multisig `Send` ignores any return value: https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/multisig/multisig_actor.go#L311

Reply:
You are quite right, thank you.

We should later change `Approve` to return the inner return value (but then to check that we'll have to use signers >1).